### PR TITLE
implement scratch ListOperations version. General upserts into DB

### PR DESCRIPTION
**Pull request by qrort**

### CODE REVIEW

File: internal/connectors/db/process_result_set.go [hunk 1]
```diff
@@ -1,31 +1,69 @@
...
+		named.Optional("s3_region", &s3region),
+		named.Optional("s3_path_prefix", &s3pathprefix),
+		named.Optional("status", &status),
 	)
 	if err != nil {
 		return nil, err
 	}
-	return &types.Backup{ID: backupId}, nil
+
+	id, err := uuid.FromBytes(backupId[:])
```
Comment (qrort) [inline internal/connectors/db/process_result_set.go:51]: There is a strange behaviour. I use nbstream ydb docker:
image
see `id` and `status` field displays. `status` is broken, `id` is correct.
But when i query `id` via SDK, it reads following values:
```
qrort@qrort-Latitude-5511:~/ydbcp$ ./test.sh ListBackups
{
  "backups": [
    {
      "id": "c830d44f-c000-b480-11d1-9dad6ba7b810",
      "status": "PENDING"
    },
    {
      "id": "78563412-7856-3412-5678-123412345678"
    }
  ],
  "nextPageToken": "2"
}
```
The bytes get mixed somehow. I tried different convert methods, but to no avail.

File: cmd/ydbcp/main.go [hunk 2]
```diff
@@ -42,17 +44,34 @@ type server struct {
...
 
 // GetBackup implements BackupService
-func (s *server) GetBackup(ctx context.Context, in *pb.GetBackupRequest) (*pb.Backup, error) {
-	log.Printf("Received: %v", in.GetId())
-	backups, err := s.driver.SelectBackupsByStatus(ctx, types.BackupStatePending)
+func (s *server) GetBackup(ctx context.Context, request *pb.GetBackupRequest) (*pb.Backup, error) {
+	log.Printf("Received: %v", request.GetId())
+	requestId, err := types.ParseObjectId(request.GetId())
+	if err != nil {
+		xlog.Error(ctx, "Failed to parse UUID", zap.Error(err))
```
Comment (bma13) [inline cmd/ydbcp/main.go:51]: `return nil, fmt.Errorf("failed to parse uuid %s: %w", request.GetId(), err)`

File: cmd/ydbcp/main.go [hunk 3]
```diff
@@ -42,17 +44,34 @@ type server struct {
...
 	if err != nil {
 		xlog.Error(ctx, "can't select backups", zap.Error(err))
 		return nil, err
 	}
-	for _, backup := range backups {
-		fmt.Println("backup:", backup.ID.String())
+	if len(backups) == 0 {
+		return nil, errors.New("No backup with such Id")
+	} else {
+		fmt.Println("backup:", backups[0].ID.String())
```
Comment (bma13) [inline cmd/ydbcp/main.go:72]: xlog.Debug()?

File: cmd/ydbcp/main.go [hunk 4]
```diff
@@ -42,17 +44,34 @@ type server struct {
 }
 
 // GetBackup implements BackupService
-func (s *server) GetBackup(ctx context.Context, in *pb.GetBackupRequest) (*pb.Backup, error) {
-	log.Printf("Received: %v", in.GetId())
-	backups, err := s.driver.SelectBackupsByStatus(ctx, types.BackupStatePending)
+func (s *server) GetBackup(ctx context.Context, request *pb.GetBackupRequest) (*pb.Backup, error) {
+	log.Printf("Received: %v", request.GetId())
```
Comment (bma13) [inline cmd/ydbcp/main.go:48]: xlog.Info ?

File: cmd/ydbcp/main.go (hunk 3 above)
Comment (bma13) [reply cmd/ydbcp/main.go:72]: Do we need 'else' block ?

File: internal/connectors/db/process_result_set.go [hunk 5]
```diff
@@ -50,7 +88,7 @@ func ReadOperationFromResultSet(res result.Result) (types.Operation, error) {
 	}
 	if operationType == types.OperationTypeTB {
 		if backupId == nil || database == nil || ydbOperationId == nil {
-			return nil, fmt.Errorf("failed to read required fields of operation %s", operationId.String())
+			return nil, errors.New(fmt.Sprintf("failed to read required fields of operation %s", operationId.String()))
```
Comment (bma13) [inline internal/connectors/db/process_result_set.go:91]: What's the point of using `errors.New(fmt.Sprintf())` instead of `fmt.Errorf()` ?

**Review by bma13 — APPROVED** [review]

**Pull request merged**